### PR TITLE
Feature: Add WithdrawWatcher Job

### PR DESCRIPTION
**Pull request by dpatsora**

### CODE REVIEW

File: app/jobs/cron/withdraw_watcher.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  module Cron
+    module WithdrawWatcher
+      class << self
+        def process
+          process_under_review_withdrawals
+          process_confirming_withdrawals
+          sleep 25
+        end
+
+        # Process withdrawals with `under_review` state each minute
```
Comment (dpatsora) [inline app/jobs/cron/withdraw_watcher.rb]: remove

File: app/jobs/cron/withdraw_watcher.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            next unless wallet.gateway_implements?(:fetch_blockchain_transaction_id)
+
+            begin
+              configure_service_adapter(withdraw)
+              fetch_withdraw_txid(withdraw)
+              Rails.logger.warn "Txid for withdraw #{withdraw.id} is not available" if withdraw.txid.nil?
+            rescue StandardError => e
+              Rails.logger.error { "Failed to fetch txId for withdraw #{withdraw.id}. See exception details below." }
+              report_exception(e)
+              # Maybe we can skip this
```
Comment (dpatsora) [inline app/jobs/cron/withdraw_watcher.rb]: remove

File: app/models/wallet.rb [hunk 3]
```diff
@@ -43,6 +43,7 @@ class Wallet < ApplicationRecord
 
   validates :name,    presence: true, uniqueness: true
   validates :address, presence: true
+  validate :gateway_wallet_kind_support
```
Comment (dpatsora) [inline app/models/wallet.rb:46]: @calj are you ok with such naming?

**Review by calj — APPROVED** [review]

File: app/workers/daemons/deposit.rb [hunk 4]
```diff
@@ -78,7 +78,7 @@ def process_deposit(deposit)
       def collect_fee(deposit)
         if deposit.spread.blank?
           deposit.spread_between_wallets!
-          Rails.logger.warn { "The deposit was spreaded in the next way: #{deposit.spread}"}
```
Comment (mnaichuk) [inline app/workers/daemons/deposit.rb:81]: gateway_implements? use in this file also

File: app/jobs/cron/withdraw_watcher.rb [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+            wallet = Wallet.active.joins(:currencies)
+                           .find_by(currencies: { id: withdraw.currency_id }, kind: :hot)
+
+            unless wallet
+              Rails.logger.warn { "Can't find active hot wallet for currency with code: #{withdraw.currency_id}." }
+              next
+            end
+
+            @service = WalletService.new(wallet)
```
Comment (mnaichuk) [inline app/jobs/cron/withdraw_watcher.rb:32]: please remove this

**Review by mnaichuk — APPROVED** [review]

**Pull request merged**